doc: add common suggestions (#134)
operator-framework/olm-docs: content/en/docs/best-practices/images/common.md

---
title: "Common recommendations and suggestions"
linkTitle: "Common suggestions"
weight: 2
description: Common recommendations and suggestions to built solutions with Operator OLM
---

## Overview

Any recommendation or good practice suggested by the Kubernetes community such as to develop [Operator pattern][operator-pattern] solutions or to manage them are good recommendations for who is looking for to build the operator projects and distribute them with OLM. Also, see [Operator Best Practices][operator-best-practices].

## Provide what are the k8s versions supported by your project

> In this CSV manifest, you will check that it provides the spec field `minKubeVersion`. Then, would be recommended you provide this information. Otherwise, it would mean that your operator project can be distributed and installed in any cluster version supported by OLM (>= `v1.11.3+) which not necessary is the case.

Reviewer: Looks like this is missing a trailing formatter.

```suggestion
In this CSV manifest, you will check that it provides the spec field `minKubeVersion`. Then, would be recommended you provide this information. Otherwise, it would mean that your operator project can be distributed and installed in any cluster version supported by OLM (>= `v1.11.3+`) which not necessary is the case.
```

## Validate your bundle before publish it

> Check and test your operator bundle before you publish it. Note that [`operator-sdk`][operator-sdk] CLI can help with. You can validate a bundle via  [`operator-sdk bundle validate`][sdk-cli-bundle-validate] against the entire suite of validators for Operator Framework, in addition to required bundle validators:

Reviewer: ```suggestion
Check and test your operator bundle before you publish it. Note that [`operator-sdk`][operator-sdk] CLI can help with that process. You can validate a bundle via  [`operator-sdk bundle validate`][sdk-cli-bundle-validate] against the entire suite of validators for Operator Framework, in addition to required bundle validators:
```

```sh
operator-sdk bundle validate ./bundle --select-optional suite=operatorframework
```

> Also, see that you can use validate a bundle via  [`operator-sdk scorecard`][sdk-cli-scorecard-bundle] to ensure it against a suite of tests:

Reviewer: ```suggestion
Also,  you can validate a bundle via [`operator-sdk scorecard`][sdk-cli-scorecard-bundle] to insure it against a suite of tests:
```
